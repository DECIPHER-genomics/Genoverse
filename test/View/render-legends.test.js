const { afterTest, testTrackRenderStatic } = require('../utils');

describe('Correctly render legends:', () => {
  afterEach(afterTest);

  const track = {
    id            : 'test',
    margin        : 0,
    featureHeight : 10,
    featureMargin : {},
    legend        : true,
  };

  it('multiple feature types', () => testTrackRenderStatic(
    [{ start: 1, end: 5, color: 'red', legend: 'test1' }, { start: 6, end: 10, color: 'blue', legend: 'test2' }, { start: 11, end: 15, color: 'green', legend: 'test3' }],
    track,
    { width: 1000 }
  ));

  it('legends are sorted alphabetically, regardless of feature order', () => testTrackRenderStatic(
    [{ start: 1, end: 5, color: 'red', legend: 'zzz' }, { start: 6, end: 10, color: 'blue', legend: 'aaa' }, { start: 11, end: 15, color: 'green', legend: 'bbb' }],
    track,
    { width: 1000 }
  ));

  it("feature doesn't have legend property", () => testTrackRenderStatic(
    [{ start: 1, end: 5, color: 'red', legend: 'zzz' }, { start: 6, end: 10, color: 'blue', legend: 'aaa' }, { start: 11, end: 15, color: 'green' }],
    track,
    { width: 1000 }
  ));
});
