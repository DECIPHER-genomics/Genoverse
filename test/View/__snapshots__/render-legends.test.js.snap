// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly render legends: feature doesn't have legend property 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAA+gAAAAMCAYAAADxnpftAAAABmJLR0QA/wD/AP+gvaeTAAAAcElEQVR4nO3XwQ0AIAwDsZbJ2RymQESVPUE+eVyfqlPAaO3mMN/u3wuA13Y5Ogy3fg8AAAAABDoAAABEEOgAAAAQQKADAABAAIEOAAAAAQQ6AAAABBDoAAAAEECgAwAAQACBDgAAAAAAAAAAAABAiAu6dQQUEkxCZQAAAABJRU5ErkJggg=="`;

exports[`Correctly render legends: feature doesn't have legend property 2`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAA+gAAAAUCAYAAAAeGxcbAAAABmJLR0QA/wD/AP+gvaeTAAABe0lEQVR4nO3drapUURjH4d/41QSxWr0Cm8ViFizaBJOI2W46GLwJ4USDUbwA8QIsBs0img2WbZg55TAjM/vsM3vE54ENa//Lu9pai3d/FAAAAAAAAAAAAAAAAAAAAAAAAAAHY7E+HoaJy2yoAwCcl6EmXc8XG/cNAMAULsw9AQAAAMABHQAAAA6CAzoAAAAcgEsz139YPa+uVN+qR9XPETkAMJ8H1ZPV+HrLd9Vfbpnd2t80AeCfNAzTXhudLM5Vj6ujkTkAcMrEi/k2H5y7WL1veWDfNQOA/97cHfQb1XF1tbpcfR+ZAwDzO6o+VW9GZADAenvroH+s7qzGt6t3I3MA4JQ9d9DvVR9avoa2awYANH8H/Vr1eTW+f4YcAJjPzepVdbf6vWMGAKzMfUB/Ub2tvlQ/zpADAPN51rIj/np1/7X6tWX2dH/TBIDDtlgf//Wx9DFlNtQBAM7Llh9229pi474BAJiC/6ADAAAAAAAAAAAAAAAAAABw4g+2Q+eD/eM8mAAAAABJRU5ErkJggg=="`;

exports[`Correctly render legends: legends are sorted alphabetically, regardless of feature order 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAA+gAAAAMCAYAAADxnpftAAAABmJLR0QA/wD/AP+gvaeTAAAAcElEQVR4nO3XwQ0AIAwDsZbJ2RymQESVPUE+eVyfqlPAaO3mMN/u3wuA13Y5Ogy3fg8AAAAABDoAAABEEOgAAAAQQKADAABAAIEOAAAAAQQ6AAAABBDoAAAAEECgAwAAQACBDgAAAAAAAAAAAABAiAu6dQQUEkxCZQAAAABJRU5ErkJggg=="`;

exports[`Correctly render legends: legends are sorted alphabetically, regardless of feature order 2`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAA+gAAAAjCAYAAAAHZCAOAAAABmJLR0QA/wD/AP+gvaeTAAACPUlEQVR4nO3dsYtUVxjG4d+q0cogIdhsK3aBaGNnIKkDIqj/gYhdQnqnM1YpBSshlVXKEAgkzTaCIFZbaBsIKYWUTopZC2En7N6dnXvdeR4YmHmb73R3Ps453y0AAAAAAAAAAAAAAAAAAAAAAAAAJmNr/3g+X3GZJXUAgGMza7XP89my/w0AwCqcGnsBAAAAgAYdADi6x9WNFeQAsNE06AAAADABYzfot6vn1cvq1+rzgTkAMK6vqj+q3ermEXIA2FhjN+i/V9eqL6tn1XcDcwBgXJerb6qvq5+qTwfmALCxzoxcf7v6uTpffVL9PTAHAMb1rHpX/VW9qL4YkO+scb0AMDlj76A/qR5V16vvq3MDcwBgOpa9ju2wOQBslLEb9Ast7p7Vh9NcD5sDAOO6U51ucdrtSvVqYA4AG2vsI+4Pql+q19U/R8gBgHHtVn9WF1vMiHk7MAeAjbXkSNl8vuIyjq4BwLrNWu3zfOYoOgAcp7GPuAMAAAAAAAAAAAAAAAAAADAd+w57mbfaoTJb3m8KAAAA/8uQOAAAAJgADToAAABMgAYdAAAAJuDMyPVvVXf3vn/W4q76wwNmV9e3TAAAABjBvOar/Byg5OnqtxYN+2EzAAAA+OhNZYr7jy12838YkAEAAMDJtOYd9G+rnersgAwAAABOrjU26Jeq3Wp7QAYAAAAnxthD4u632BF/uvf7TfXvAbN761smAAAAHK+p3EEHAACAjeY96AAAAAAAAAAAAAAAAAAAALz3H6cw6OV7O5JsAAAAAElFTkSuQmCC"`;

exports[`Correctly render legends: multiple feature types 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAA+gAAAAMCAYAAADxnpftAAAABmJLR0QA/wD/AP+gvaeTAAAAcElEQVR4nO3XwQ0AIAwDsZbJ2RymQESVPUE+eVyfqlPAaO3mMN/u3wuA13Y5Ogy3fg8AAAAABDoAAABEEOgAAAAQQKADAABAAIEOAAAAAQQ6AAAABBDoAAAAEECgAwAAQACBDgAAAAAAAAAAAABAiAu6dQQUEkxCZQAAAABJRU5ErkJggg=="`;

exports[`Correctly render legends: multiple feature types 2`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAA+gAAAAjCAYAAAAHZCAOAAAABmJLR0QA/wD/AP+gvaeTAAAD7klEQVR4nO3dTahmcxzA8e+MEeOlTHmJmhhkYSOLUSyQRtmMQfaSLLwUCxvZ3EmyQLKlWBAp5W3jZQopCYm8LBRKktgoFiPlWpzn1nR7Zpi553m5+nzq6d5zzvM75/c8m3+/8/s/518AAAAAAAAAAAAAAAAAAAAAAAAALI0t03au1urIF5l6HQBgllZHHc9ri/EcAGZo66ITAAAAABZfoG+vbhsx9rnq1+qLjSQFAByVMcfzndWB6sfq2+rujaUGAJvc6jAnbrTXES51esdeTE+LvbravYFzAsD/yKjD+bzG853VlZP/z2wo1C8+xnMDwOY3xwL98er36u1q/2TfzdVH1afVs9UJ1Y6Gu+lfVZ9VNx4mturCFOgA0BwL9FmM52sOVHs2/l0AwCa1wA76ruqd6sTJ9v6GqW23VI8d8r5Tp8SuUaADQLXADvoY43kNnfPvqlOO4cMDwKazbdEJrHNVdW71+mT7pOq96oVqpfq7ern6oOFOPACwfMYYz3dM3n9n9ccskwWAZbFsBfrW6s3qjinHdld7q0erV6qn55gXAPDfbXQ83z459kT1xoxyBIDNYY5T3E+sfjlk+/zqp8nfGu6eX9DwwJi1O+x7G+6or49dY4o7AFRznOI+5ni+rXq1un/jnx8A/gfmWKBXPVV903AnvWpf9Xn19eTv1dUNDUX3l9Un1SWHiX2x+qH6s2FplltH+UIAYFOaW4Fe443n1zRMgf/2kNe+sb4RAFhmW6bt/A9F9dFeZOp1AIBZ+tei+ihtMZ4DwAxtXXQCAAAAAAAAAAAAAAAAAAAALI3pD3tZGfchca14SBwAAAAciYfEAQAAwBJYdIG+vbptpNjjqo+r7xvWQn8sy7sBAACwSSy6QD+5unek2L+rvdWu6qLq8uq6DWUHAAAAc7LoAv2B6rzq7Wr/ZN/N1UfVp9Wz1QnVjupA9VX1WXXjlNjV6ufJOY5r+Gzj/pYeAAAAZmTbgq//ULWnunayvau6q7qyOthQeN9e/V59Pnlv1anV++ti13xT7ayeq96aYe4AAAAwmkV30Ne7qjq3er2hM76nOqehm35T9Uh1RUPBfjgXVWdXF1SXzTJZAAAAGMuiO+jrba3erO6Ycmx3w2/MH61eqZ4+wnl+m5xnb/XhyDkCAADA6BbdQf+jOuuQ7XerfdX5k+0dDZ3wnQ1d82eqh6tLp8Se1dB9X4u7vvp6VokDAADAmBbdQT9Yvdrwu/HXqvsauucvV8dXf1X3VKdVDzYsm3awYXm19bFPVi9VZ0yOPT95AQAAwNKbvk74yshPP1+xHjkAAAAcyaKnuAMAAAAAAAAAAAAAAAAAALA8/gF/eUZXKd+oegAAAABJRU5ErkJggg=="`;
